src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -239,10 +241,34 @@ public static long countBulkFileLoadModelsFromPath(
             cloudEncapsulationId,
             fileRefColumns,
             errors)) {
+      nodesStream
+          .takeWhile(bulkLoadFileModel -> errors.size() <= maxBadLoadFileLineErrorsReported)
+          .forEach(
+              loadFileModel -> {
+                try {
+                  cloudFileReader.validateUserCanRead(
+                      List.of(loadFileModel.getSourcePath()), userRequest);
+                } catch (BlobAccessNotAuthorizedException ex) {
+                  recordControlFileValidationErrorMessage(
+                      errors, ex.getMessage(), maxBadLoadFileLineErrorsReported);
+                }
+              });
       return nodesStream.count();
     }
   }
 
+  public static void recordControlFileValidationErrorMessage(
+      List<String> errors, String exceptionMsg, int maxBadLoadFileLineErrorsReported) {
+    if (errors.size() < maxBadLoadFileLineErrorsReported) {
+      errors.add("Error: " + exceptionMsg);
+    } else {
+      errors.add(
```

Comment:
Won't this cause the truncated message to be added multiple times? It seems like you'd want it to only be added once. Another approach would be to allow this method to append any number of times and truncate in the caller.

Reply:
Another approach, which may be too clever, would be to check `errors.size() ==  maxBadLoadFileLineErrorsReported` in the `else`.

Reply:
Is the code is relying on `takeWhile()` to do this check? I don't know if you can rely on the stream to execute `peek()` and then re-evaluate `takeWhile()` after each call.

Could you combine the `peek()` into the `takeWhile()`? That could work.
```
      return nodesStream
          .takeWhile(bulkLoadFileModel -> {
                try {
                  cloudFileReader.validateUserCanRead(
                      List.of(bulkLoadFileModel.getSourcePath()), userRequest);
                } catch (BlobAccessNotAuthorizedException ex) {
                  recordControlFileValidationErrorMessage(
                      errors, ex.getMessage(), maxBadLoadFileLineErrorsReported);
                }
                return errors.size() <= maxBadLoadFileLineErrorsReported;
              })
          .count();
```

Reply:
Or
```
      return nodesStream
          .takeWhile(bulkLoadFileModel -> {
                try {
                  cloudFileReader.validateUserCanRead(
                      List.of(bulkLoadFileModel.getSourcePath()), userRequest);
                } catch (BlobAccessNotAuthorizedException ex) {
                  if (errors.size() < maxBadLoadFileLineErrorsReported) {
                    errors.add("Error: " + ex.getMessage());
                  } else {
                    errors.add(
                        "Error details truncated. [MaxBadLoadFileLineErrorsReported = "
                            + maxBadLoadFileLineErrorsReported
                            + "]");
                    return false;
                  }
                }
                return true;
              })
          .count();
```

I'd like to use `!=` here instead of `<` but the way it's written this code can't be sure that `errors` doesn't already contain some data. It might be clearer to have this return both the count and the errors so that it's clear that the errors are generated here, and no errors are passed in.

Reply:
[discussed in conversations offline] 
But just to note for other reviewers:
- I'm now back to just using peek() b/c I no longer need the check that we've hit the max number of errors
- Rather than returning true/false, I'm throwing an exception once we've hit the max errors.